x-pack/plugins/observability_solution/apm/server/routes/historical_data/has_historical_agent_data.ts

```diff
@@ -23,8 +24,10 @@ export async function hasHistoricalAgentData(apmEventClient: APMEventClient) {
   return hasDataUnbounded;
 }
 
-type DataTier = 'data_hot' | 'data_warm' | 'data_cold' | 'data_frozen';
-async function hasDataRequest(apmEventClient: APMEventClient, dataTiers?: DataTier[]) {
+async function hasDataRequest(
+  apmEventClient: APMEventClient,
+  dataTiers?: IndexLifeCycleDataTier[]
```

Comment:
does this still work? does it override the setting?

Reply:
Ok that makes sense, should we add a comment? e.g.: "if the user has disabled querying the frozen tier, this will take precedence over `dataTiers`". Just to make it more explicit. The other question I have: can we clearly mark this setting as tech preview, so we can get rid of it later with a breaking change? (I'm assuming there will be a kibana-wide setting at some point).

Reply:
yeah. it makes sense. I'll add a comment.